File: 70. Climbing Stairs/climbing-stairs.c
// 70. Climbing Stairs (1/11/2022)
// Runtime: 0 ms (91.27%) Memory: 5.72 MB (0.00%) 

int climbStairs(int n) {
    if (n <= 2) {
        return n;
    }

    int64_t prev_1 = 2, prev_2 = 1, now = 0;

    for (int64_t i = 3; i <= n; i++) {
        now = prev_1 + prev_2;
        prev_2 = prev_1;
        prev_1 = now;
    }

    return now;
}

File: 70. Climbing Stairs/climbing-stairs.js
// 70. Climbing Stairs (12/29/53742)
// Runtime: 78 ms (39.76%) Memory: 38.06 MB (92.87%) 

/**
 * @param {number} n
 * @return {number}
 */
function climbStairs(n) {
    const dp = [1, 2];
    for (let i = 2; i < n; i++) dp[i] = dp[i - 1] + dp[i - 2];
    return dp[n - 1];
}